ui/app/pages/confirm-transaction/conf-tx.js

```diff
@@ -13,20 +13,30 @@ import Loading from '../../components/ui/loading-screen'
 import { DEFAULT_ROUTE } from '../../helpers/constants/routes'
 
 function mapStateToProps (state) {
-  const { metamask } = state
+  const { metamask, appState } = state
   const {
+    network,
     unapprovedMsgCount,
+    unapprovedMsgs,
     unapprovedPersonalMsgCount,
+    unapprovedPersonalMsgs,
+    unapprovedTxs,
+    unapprovedTypedMessages,
     unapprovedTypedMessagesCount,
   } = metamask
+  const {
+    txId,
+  } = appState
 
   return {
     identities: state.metamask.identities,
     unapprovedTxs: state.metamask.unapprovedTxs,
     unapprovedMsgs: state.metamask.unapprovedMsgs,
     unapprovedPersonalMsgs: state.metamask.unapprovedPersonalMsgs,
     unapprovedTypedMessages: state.metamask.unapprovedTypedMessages,
-    index: state.appState.currentView.context,
+    index: txId
+      ? txHelper(unapprovedTxs, unapprovedMsgs, unapprovedPersonalMsgs, unapprovedTypedMessages, network).findIndex((tx) => tx.id === txId)
```

Comment:
This change seems unnecessary. Why not just use `txId`? That appears to be what it did before, functionally.

Reply:
I'll try this